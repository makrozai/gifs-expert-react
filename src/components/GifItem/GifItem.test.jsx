import { render, screen } from "@testing-library/react";
import { GifItem } from "./GifItem"

describe('Testing <GifItem />', () => {
  
  const title = "Imagen de prueba"
  const url = "http://google.com/prueba.png"
  
  test('Evaluar snapshot', () => {

    const { container } = render(<GifItem title={title} url={ url }  />)
    expect(container).toMatchSnapshot()

  })

  test('Debe mostrar la imagen con el URL y el ALT indicado', () => {
    
    render(<GifItem title={title} url={url} />)

    const { src, alt } = screen.getByRole('img')
    expect(src).toBe(url)
    expect(alt).toBe(title)

  })

  test('Debe de mostrar el titulo en el componente', () => {

    render(<GifItem title={title} url={url} />)
    expect(screen.getByText(title)).toBeTruthy()
    
  })
  
})