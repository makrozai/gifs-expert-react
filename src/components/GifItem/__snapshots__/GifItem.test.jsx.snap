// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <GifItem /> Evaluar snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Imagen de prueba"
      src="http://google.com/prueba.png"
    />
    <p>
      Imagen de prueba
    </p>
  </div>
</div>
`;
